### DISCUSSION TITLE
fails in GitHub Actions

### DISCUSSION THREAD

**Discussion by sajis997** [obi1kenobi/cargo-semver-checks#722] [Q&A]
usiing the following in the pipeline

    - name: Check semver
      uses: obi1kenobi/cargo-semver-checks-action@v2

I am getting the following error:

error: no crates with library targets selected, nothing to semver-check
Error: The process '/home/runner/.cargo/bin/cargo' failed with exit code 1

**Comment by obi1kenobi** [comment]
Hi! Thanks for checking out cargo-semver-checks.

This error indicates that the project that is being scanned is not a library. It might be a binary crate (e.g. it has a `main.rs` instead of `lib.rs`), or another type of crate such as a proc-macro crate or something else. Since there is no library API, there's nothing that cargo-semver-checks can run any kind of check on.

Could you confirm whether your project has a library component? If you still feel this is a bug in cargo-semver-checks, I'd love to know what project this is in so I can try to reproduce it and figure out what's going on.

Thanks!

**Reply by sajis997** [reply]
the project is not a library , it is a binary project but uses several library crate as follows:

assert_approx_eq = "1.1.0"
rand = "0.8.5"
image = {version = "0.25.0", features = ["rayon"] }
rayon = "1.10.0"
indicatif = { version = "0.17.8", features = ["rayon"] }
glam = "0.27.0"
tobj = "4.0.2"

**Reply by obi1kenobi** [reply]
In this case, could you let me know what part of this project you're hoping to check with cargo-semver-checks?

For example, could you describe an error that you'd like cargo-semver-checks to catch if it happens in your binary project?
